packages/app/src/api/visualization.js

```diff
@@ -1,46 +1,36 @@
-import { getInstance } from 'd2'
 import { getFieldsStringByType } from '../modules/fields'
 
-export const apiFetchVisualization = (type, id) =>
-    getInstance().then(d2 =>
-        d2.models[type].get(id, {
-            fields: getFieldsStringByType(type),
-        })
-    )
+export const apiFetchVisualization = (dataEngine, id) => {
+    const visualizationQuery = {
+        visualization: {
+            resource: 'visualizations',
+            id,
+            params: {
+                fields: getFieldsStringByType('visualization'),
+            },
+        },
+    }
 
-export const apiSaveVisualization = (type, visualization) =>
-    getInstance()
-        .then(d2 => d2.models[type])
-        .then(modelDefinition => {
-            const api = modelDefinition.api
-            const apiEndpoint = modelDefinition.apiEndpoint
+    return dataEngine.query(visualizationQuery)
+}
 
-            const options = {
-                skipTranslations: true,
-                skipSharing: true,
-            }
+export const apiSaveVisualization = (dataEngine, visualization) => {
+    const mutation = {
```

Comment:
Since this mutation has a dynamic type and ID it can't be extracted, unfortunately, so this is probably the best way to do it here